File: libs/src/main/java/com/mmgsoft/modules/libs/amzbiling/AmazonIapActivity.java
package com.mmgsoft.modules.libs.amzbiling;

import androidx.appcompat.widget.AppCompatImageView;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.amazon.device.iap.PurchasingService;
import com.mmgsoft.modules.libs.R;
import com.mmgsoft.modules.libs.R2;
import com.mmgsoft.modules.libs.customview.SpacesItemDecoration;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

import butterknife.BindView;

public class AmazonIapActivity extends BaseIapAmzActivity implements OnItemClickListener<ProductItem> {

    @BindView(R2.id.imBack)
    public AppCompatImageView imBack;

    @BindView(R2.id.rcvSkuAvai)
    public RecyclerView rcvSkuAvai;

    private ProductSkuAdapter mSkuAdapter;


    @Override
    public Set<String> getAllSkus() {
        Set<String> listSkus = new HashSet<>();
        listSkus.addAll(Arrays.asList(AppConstant.ITEM_ENTITLE, AppConstant.ITEM_SUB_TERM_1));
        return listSkus;
    }

    @Override
    public Integer getResLayout() {
        return R.layout.purchase_product_layout;
    }

    @Override
    public void initData() {
        imBack.setOnClickListener(v -> {
            super.onBackPressed();
        });
        initRecycler();
    }

    @Override
    public void notifyUpdateListView() {
        mSkuAdapter.notifyDataSetChanged();
    }

    private void initRecycler() {
        mSkuAdapter = new ProductSkuAdapter(this);
        rcvSkuAvai.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        rcvSkuAvai.setItemAnimator(new DefaultItemAnimator());
        rcvSkuAvai.addItemDecoration(new SpacesItemDecoration(10, true));
        rcvSkuAvai.setHasFixedSize(true);
        rcvSkuAvai.setAdapter(mSkuAdapter);
        mSkuAdapter.updateAllData(productItems);
    }

    @Override
    public void onItemClicked(ProductItem item) {
        PurchasingService.purchase(item.sku);
    }
}

File: libs/src/main/java/com/mmgsoft/modules/libs/amzbiling/AmazonIapActivity.kt
package com.mmgsoft.modules.libs.amzbiling

import android.content.Context
import android.content.Intent
import android.view.View
import androidx.appcompat.widget.AppCompatImageView
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.amazon.device.iap.PurchasingService
import com.mmgsoft.modules.libs.R
import com.mmgsoft.modules.libs.customview.SpacesItemDecoration
import com.mmgsoft.modules.libs.helpers.AmazonScreenType
import com.mmgsoft.modules.libs.utils.AdsComponentConfig.amazonProdId

class AmazonIapActivity : BaseIapAmzActivity(), OnItemClickListener<ProductItem> {
    private val imBack: AppCompatImageView by lazy {
        findViewById(R.id.imBack)
    }

    private val rcvSkuAvai: RecyclerView by lazy {
        findViewById(R.id.rcvSkuAvai)
    }

    private val mSkuAdapter: ProductSkuAdapter by lazy {
        ProductSkuAdapter(this)
    }

    override val allSkus: Set<String>
        get() = HashSet(amazonProdId)

    override val resLayout: Int
        get() = R.layout.purchase_product_layout

    companion object {
        const val EXTRA_SCREEN_TYPE = "EXTRA_SCREEN_TYPE"

        fun open(ctx: Context, screenType: AmazonScreenType = AmazonScreenType.SUBSCRIPTION) =
            Intent(ctx, AmazonIapActivity::class.java).apply {
                putExtra(EXTRA_SCREEN_TYPE, screenType.type)
                ctx.startActivity(this)
            }
    }

    override val screenType: AmazonScreenType
        get() = AmazonScreenType.map(
            intent.getStringExtra(EXTRA_SCREEN_TYPE) ?: AmazonScreenType.BUY_GOLD.type
        )

    override fun initData() {
        imBack.setOnClickListener { v: View? -> super.onBackPressed() }
        initRecycler()
    }

    override fun notifyUpdateListView() {
        mSkuAdapter.notifyDataSetChanged()
    }

    private fun initRecycler() {
        rcvSkuAvai.layoutManager = LinearLayoutManager(
            this,
            LinearLayoutManager.VERTICAL,
            false
        )
        rcvSkuAvai.itemAnimator = DefaultItemAnimator()
        rcvSkuAvai.addItemDecoration(SpacesItemDecoration(10, true))
        rcvSkuAvai.setHasFixedSize(true)
        rcvSkuAvai.adapter = mSkuAdapter
        mSkuAdapter.updateAllData(productItems)
    }

    override fun onItemClicked(item: ProductItem) {
        PurchasingService.purchase(item.sku)
    }
}